test/index.js

```diff
@@ -1,88 +1,128 @@
 import test from 'tape'
-import { encode, decode, validate } from '../index.js'
-import { randomBytes } from 'node:crypto'
-import { docs } from './docs.js'
+import { fixtures } from './fixture.js'
+import { encode } from '../dist/encode.js'
+import { decode } from '../dist/decode.js'
+import { parseVersionId } from '../dist/lib/utils.js'
+import { and } from 'ajv/dist/compile/codegen/index.js'
+// import { randomBytes } from 'node:crypto'
 
-test('test encoding of record with missing fields', async (t) => {
-  t.plan(1)
-  t.throws(() => {
-    encode(docs.onlyId)
-  })
-})
-
-test('test encoding of wrong record type', async (t) => {
-  t.plan(1)
-  t.throws(() => {
-    encode(docs.badDocType)
+{
+  const {text,plan,doc} = fixtures.onlyId
+  test(text, async(t) => {
+    t.plan(plan)
+    t.throws(() => {
+      encode(doc)
+    })
   })
-})
+}
 
-test('test encoding of record with wrong schema version', async (t) => {
-  t.plan(1)
-  t.throws(() => {
-    encode(docs.badSchemaVersion)
+{
+  const {text,plan, doc} = fixtures.badDocName
+  test(text,async(t) => {
+    t.plan(plan)
+    t.throws(() => {
+      encode(doc)
+    })
   })
-})
+}
 
-test('test encoding of rightfully formated record', async (t) => {
-  const goodDocs = Object.keys(docs.good)
-  t.plan(goodDocs.length)
-  goodDocs.forEach((k) => {
-    const doc = docs.good[k]
-    t.doesNotThrow(() => {
+test('testing rightfully encoding of doc',async(t) => {
+  fixtures.goodDocs.forEach(
+    /** @param {import('../dist/types').MapeoDoc} doc */
+    (doc) => {
+    t.doesNotThrow(() =>  {
       encode(doc)
-    }, `testing ${k}`)
+    }, `testing ${doc.schemaName}`)
   })
 })
 
-test('test validation of record', async (t) => {
-  const goodDocs = Object.keys(docs.good)
-  t.plan(goodDocs.length)
-  goodDocs.forEach((k) => {
-    const doc = docs.good[k]
-    const record = decode(encode(doc), {
-      coreKey: randomBytes(32),
-      index: 0,
+test('testing encoding of doc, then decoding and comparing the two objects', async(t) => {
+  fixtures.goodDocs.forEach(
+    /** @param {import('../dist/types').MapeoDoc} doc */
+    (doc) => {
+      t.doesNotThrow(() =>  {
+        const buf = encode(doc)
+        const decodedDoc = stripUndef(decode(buf, parseVersionId(doc.versionId)))
+        t.deepEqual(doc,decodedDoc)
```

Comment:
```suggestion
        t.deepEqual(decodedDoc, doc)
```

Function signature is `t.deepEqual(actual, expected)`.

Reply:
Move this outside the `doesNotThrow` callback, to be clear about what we are testing.